Fixing the CRM Skeleton Input

christie94h opened the pull request:
As discussed with @xlai earlier and outlined in #45, the input for the skeleton in Trial Designs was difficult and not consistent with the number of doses. To compensate, I've changed the input to a reactive editable table which dynamically changes length with the number of doses and outputs a shared vector which is used in the CRM simulations. 

For now, the default inputs in the table are a basic increasing sequence of equal separation, which will be changed to a much reasonable function in the future.

### Testing this PR

1. Go to Trial Designs and change the number of doses (general parameters). 
2. Go to the CRM parameters and scroll down to the input of prior dose DLTs. Does the table have the same number of rows as there are doses and is there an increasing sequence of values between 0 and 1?
3. Fill in the table. Does it save your inputs in the right place? (Note that there is no validation on this table yet, so you should be able to put anything in there for now).
4. Run a CRM simulation. The desired result should appear without crashing the app.

### Next Steps

- [ ] Add validation to the table. 
- [x] Change the sequence of default inputs so that it matches a general CRM prior.

Commit:
Changing the input of the skeleton within CRM to a table format, which varies in size with number of doses and works within the simulation code.

Commit:
Changing the default value of prior guess of MTD so that it is not more than the default number of doses.

xlai commented on app_skeleton/pages/trial_design_ui.R:
```diff
@@ -144,6 +146,41 @@ choices = c("Yes" = TRUE, "No" = FALSE), selected = TRUE, inline = TRUE)
+##### Reactive Skeleton Input ######
+reactive_skeleton <- reactiveVal() # initalising a reactive value to store the data frame
+
+  observeEvent({input$n_doses_inputt}, {
+    if (is.na(input$n_doses_inputt)) {
+      reactive_skeleton(NULL)
+    } else {
+  dose <- as.integer(input$n_doses_inputt)
+  large_vector <- c(2:(dose + 1))
+  Prior <- large_vector/(dose + 2)
```
```suggestion
  observeEvent({input$n_doses_inputt | input$prior_mtd_input}, {
    if (is.na(input$n_doses_inputt) | is.na(input$prior_mtd_input)) {
      reactive_skeleton(NULL)
    } else {
  dose <- as.integer(input$n_doses_inputt)
  Prior <- dfcrm::getprior(halfwidth = 0.25*input$ttl_inputt, target = input$ttl_inputt, nu = input$prior_mtd_input, nlevel = dose)
```

xlai replied:
This should do the trick

xlai reviewed (changes requested):
I've suggested a minor change to tick off one of your todos. Test it to see if it is reactive to both number of doses and prior mtd guess.

Validation can wait.

Commit:
Changes the skeleton so that it depends on the prior MTD.

Commit:
Fixing the crash that happens when n doses < prior mtd

christie94h commented:
@xlai I have added your suggestion to the app and it seems to work well - I added an extra line to stop the app crashing when prior mtd > n doses, sending the table to NULL. I'll add a validation message to this when I think about validation later on. 

Hopefully that's everything that needs to be done with this PR so I've re-requested a review. :)

xlai commented on app_skeleton/pages/trial_design_ui.R:
```diff
@@ -144,6 +146,42 @@ choices = c("Yes" = TRUE, "No" = FALSE), selected = TRUE, inline = TRUE)
+ observeEvent({input$n_doses_inputt | input$prior_mtd_input}, {  
+    if (is.na(input$n_doses_inputt) | is.na(input$prior_mtd_input)) {  
+      reactive_skeleton(NULL)  
+    } else if (input$prior_mtd_input > input$n_doses_inputt)  {  
+      reactive_skeleton(NULL)  
+    } else {
```
Looks great - well spotted

xlai reviewed (approved):
Everything works including the catch on invalid prior MTD guess. Approved

xlai committed:
Merge branch 'dev' into final-touch-ups-before-presentation
